tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -540,3 +544,50 @@ def test_vmgenid(guest_kernel_linux_6_1, rootfs, microvm_factory, snapshot_type)
 
         # Update the base for next iteration
         base_snapshot = snapshot
+
+
+# TODO add `global_props.host_os == "amzn2"` condition
+# once amazon linux kernels have patches.
+@pytest.mark.skipif(
+    platform.machine() != "aarch64" or global_props.host_linux_version_tpl < (6, 4),
+    reason="This is aarch64 specific test and should only be run on 6.4 and later kernels",
+)
+def test_physical_couter_reset_aarch64(uvm_nano):
+    """
+    Test that the CNTPCT_EL0 register is reset on VM boot.
+    Because we cannot read the CNTVCT_EL0 (Virtual counter)
+    from python, we assume the smallesd VM will not consume more than
+    MAX_VALUE cycles to be created and snapshotted. We have to use
```

Comment:
how was the `MAX_VALUE` chosen?

Reply:
The MAX_VALUE is retrieved from a couple of test runs with some additional buffer.